config/crd/bases/app.k8s.io_applications.yaml

```diff
@@ -72,23 +72,26 @@ spec:
                 description: AssemblyPhase represents the current phase of the application's
                   assembly. An empty value is equivalent to "Succeeded".
                 type: string
-              componentKinds:
-                description: ComponentGroupKinds is a list of Kinds for Application's
-                  components (e.g. Deployments, Pods, Services, CRDs). It can be used
-                  in conjunction with the Application's Selector to list or watch
-                  the Applications components.
+              componentGVKs:
+                description: ComponentGVKs is a list of GVKs for Application's components
+                  (e.g. Deployments, Pods, Services, CRDs). It can be used in conjunction
+                  with the Application's Selector to list or watch the Applications
+                  components.
                 items:
-                  description: GroupKind specifies a Group and a Kind, but does not
-                    force a version.  This is useful for identifying concepts during
-                    lookup stages without having partially valid types
+                  description: GroupVersionKind unambiguously identifies a kind.  It
+                    doesn't anonymously include GroupVersion to avoid automatic coersion.  It
+                    doesn't use a GroupVersion to avoid custom marshalling
                   properties:
                     group:
                       type: string
                     kind:
                       type: string
+                    version:
+                      type: string
                   required:
                   - group
                   - kind
+                  - version
```

Comment:
I think we can't make a new field (i.e. `version`) required without bumping the api version, otherwise existing Application CRs will be invalid.

Reply:
I'll make the field optional.

Reply:
I can manually update the yaml file to make the `version` field optional, but when I run `make manifests` to regenerate the CRD, the `version` field becomes required again. That is because Version is required in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/group_version.go#L86

Any better way to force it optional?